doc/architectural/compilation-and-development.md

```diff
@@ -270,3 +270,28 @@ branch, e.g. to run it on lines that have been changed from `develop`:
 
 There are also instructions for adding this as a git pre-commit hook in
 [CODING_STANDARD.md](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#pre-commit-hook-to-run-cpplint-locally).
+
+
+\section compilation-and-development-section-time-profiling Time profiling
+
+To do time profiling with a tool like `gprof`, the flags `-g` and `-pg` must be
+used which compiling, and `-pg` must be used when linking. If you are building
+with cmake you can just add "-Denable_profiling=1" to your cmake invocation, and
+reload cmake before building your desired binary.
+
+Run your binary as normal. A file called gmon.out will be created in the
+working directory of the binary at the end of execution. In most instances this
+will be the same as the working directory at the beginning of execution. It is
+also possible to choose the output location using an environment variable -
+search online for more details.
+
+Make sure gprof is installed by running `gprof -v`. If it is not installed then
+run `sudo apt install binutils`.
+
+Run `gprof <path-to-binary> <path-to-gmon.out>` and redirect the output to a
+file. This will take a while to run.
+
+The output file will now be a large text file. There are two sections: the
```

Comment:
:bulb: The official names of the two sections are "Flat profile" and "Call graph". It would be good to mention them here as their names appear as titles, i.e. are searchable, in the giant text file.
:bulb: Apparently other kinds of sections can be generated as well if `gprof` is run with certain options. Two are mentioned on https://ftp.gnu.org/old-gnu/Manuals/gprof-2.9.1/html_chapter/gprof_5.html - this or a similar tutorial might be good to link to here for people looking for more detailed info? (Again, they could google it but a link would be friendlier.)